examples/ethernet/ethernet/main/ethernet_main.c

```diff
@@ -32,92 +32,45 @@
 #include "tcpip_adapter.h"
 #include "nvs_flash.h"
 #include "driver/gpio.h"
-#include "tlk110_phy.h"
 
-static const char *TAG = "eth_demo";
+extern void phy_device_check_phy_init(void);
```

Comment:
There are a couple of functions in the eth_config structure that come from ethernet_main instead of the phy support code so a phy function like:


`    eth_config_t *eth_config = phy_device_get_config(eth_gpio_config_rmii, tcpip_adapter_eth_input);`


or the structure would not be const and ethernet_main would just set those functions.  Given that there may be more than one ethernet project in the future, the phy code should probably move out from the project and into a component support library (in the future).  That would make me lean towards the eth_config function approach.